app/models/miq_schedule.rb

```diff
@@ -4,7 +4,7 @@ class MiqSchedule < ApplicationRecord
   include_concern 'Filters'
 
   include YAMLImportExportMixin
-  deprecate_attribute :towhat, :resource_type
+  deprecate_attribute :towhat, :resource_type, :type => :string
```

Comment:
Are these the only callers of `deprecate_attribute`?  What will happen if a caller passes no/nil type?

Reply:
these are the only callers.

If there is no type, it will currently look up the type for the attribute being deprecated.

So as stated above, it is not strictly necessary.
Also, this will probably never create deadlocks because it is looking up the attribute type on the same class.

But the number of times you specifically have tracked down deadlocks due to looking up another attribute's type has me wanting to steer clear of this type of action.

Reply:
Sorry, I lost track of this.  Is this still needed?  Can you resolve the conflict if so?